### PR TITLE
executer: add `Cop_time` in slow log

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve?

Before this PR, TiDB Slow log only record `Process_time`, but it is the total process time of cop task spend on TiKV, we still don't know the related `Cop_time` spend in tidb, as below:

image

### What is changed and how it works?

Add `Cop_time` in slow_query and slow log. The `Cop_time` contain the time duration as below:
* Send Cop request.
* Send request to TiKV network latency duration.
* TiKV handle cop request duration.
* Send result to TiDB network latency.
* Recieve result latency.
* If meet error, maybe retry upper steps.

```sql
mysql>select query_time,cop_time,query from `CLUSTER_SLOW_QUERY` where cop_time>0 limit 3;
+-------------+-------------+---------------------------------------------------------------------------------------------------+
| query_time  | cop_time    | query                                                                                             |
+-------------+-------------+---------------------------------------------------------------------------------------------------+
| 0.382663125 | 0.382427668 | select count(*) from t2;                                                                          |
| 0.329472658 | 0.329190264 | select count(*) from t2;                                                                          |
| 0.929105778 | 0.400684833 | select count(*) from t2 where c2 >=0 and c1 in (select c1 from t2 where c7!=now()) and c8!=now(); |
+-------------+-------------+---------------------------------------------------------------------------------------------------+
```

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM

### CODE REVIEW

File: util/execdetails/execdetails.go [hunk 1]
```diff
@@ -180,11 +186,14 @@ func (d ExecDetails) String() string {
...
 	fields = make([]zap.Field, 0, 16)
+	if d.CopTime > 0 {
+		fields = append(fields, zap.String(strings.ToLower(CopTimeStr), strconv.FormatFloat(d.CopTime.Seconds(), 'f', -1, 64)+"s"))
+	}
 	if d.ProcessTime > 0 {
 		fields = append(fields, zap.String(strings.ToLower(ProcessTimeStr), strconv.FormatFloat(d.ProcessTime.Seconds(), 'f', -1, 64)+"s"))
 	}
 	if d.WaitTime > 0 {
-		fields = append(fields, zap.String(strings.ToLower(WaitTimeStr), strconv.FormatFloat(d.ProcessTime.Seconds(), 'f', -1, 64)+"s"))
+		fields = append(fields, zap.String(strings.ToLower(WaitTimeStr), strconv.FormatFloat(d.WaitTime.Seconds(), 'f', -1, 64)+"s"))
```
Comment (SunRunAway) [inline util/execdetails/execdetails.go:196]: Should it be fixed in an individual PR and cherry-pick it?

**Comment by SunRunAway** [comment]
I think that's a great approaching, would `cop_time` be involved in statement summary table?

File: util/execdetails/execdetails.go (hunk 1 above)
Comment (crazycs520) [reply util/execdetails/execdetails.go:196]: ok. I will fill another PR.

**Comment by crazycs520** [comment]
@SunRunAway Ye, maybe in next PR add `cop_time` to statement summary table.

**Comment by qw4990** [comment]
@crazycs520 Should we cherry pick this PR to 4.0?

**Comment by crazycs520** [comment]
@qw4990 Yes, I want to cherry-pick to 4.0.

File: util/execdetails/execdetails.go (hunk 1 above)
Comment (crazycs520) [reply util/execdetails/execdetails.go]: Fix in https://github.com/pingcap/tidb/pull/16899

**Comment by SunRunAway** [comment]
@crazycs520 we should add documentation update in `Related changes` section.

**Comment by crazycs520** [comment]
@djshow832, It's ok, new version can still read the old version tidb slow log, and old version can read the new tidb slow log, but can't read the `Cop_time` field value.

**Pull request merged**